### PR TITLE
Adds mob riding for small mobs and simple_animals.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Allows corgis and possums to be used as mounts by small mobs.

image

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Can't work out what's going on with relaymove, if anyone can advise how I can get that working with mobs please hit me up.

File: code/_helpers/mobs.dm [hunk 1]
```diff
@@ -20,8 +14,7 @@
 	for(var/atom/movable/AM in view(range, source))
 		var/M = AM.get_mob()
 		if(M)
-			mobs += M
-
+			mobs |= M
```
Comment (MistakeNot4892) [inline code/_helpers/mobs.dm:17]: `get_mob()` can return mobs that are also in view(), resulting in duplicates.

File: code/datums/movement/mob.dm [hunk 2]
```diff
@@ -1,24 +1,3 @@
-// Movement relayed to self handling
-/datum/movement_handler/mob/relayed_movement
```
Comment (MistakeNot4892) [inline code/datums/movement/mob.dm]: This appears to be a stub that Psi mentioned may date back to half-implemented borer move handling. It doesn't do anything and blocks relayed moves from mobs.

File: code/datums/movement/mob.dm [hunk 3]
```diff
@@ -102,27 +81,11 @@
 
 // Buckle movement
 /datum/movement_handler/mob/buckle_relay/DoMove(var/direction, var/mover)
-	// TODO: Datumlize buckle-handling
```
Comment (MistakeNot4892) [inline code/datums/movement/mob.dm]: Did this by splitting the logic up into a proc.

**Comment by MistakeNot4892** [comment]
This is spiralling off into a nightmare PR. Going to need to split it into several I think.

**Comment by MistakeNot4892** [comment]
Split off into #2037 and #2036

**Comment by MistakeNot4892** [comment]
Still to split out:
- ~~attack_hand refactoring for mobs~~
- grab proximity handling/afterattack stuff

**Comment by MistakeNot4892** [comment]
Currently depends on #2146 and #2145.

**Comment by MistakeNot4892** [comment]
Now that all the dependencies are in, this PR is completely broken.

**Comment by MistakeNot4892** [comment]
Depends on #2170.

**Comment by MistakeNot4892** [comment]
Vast majority of work is done but grabs are a pain to work with and I'm unlikely to find time to sort them out. Closing for now, will revive another day.

**Comment by PlayerDeer** [comment]
WOAH

**Comment by MistakeNot4892** [comment]
This is more or less working, just some UX issues to clean up and some procs to add to hook it into simplemob AI.
- Trying to click-grab with grab intent just dismounts. Ctrl click does generate the control grab.
- Control grab has no icon/icon state.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**